code/src/ng_link/ng_layer.py

```diff
@@ -0,0 +1,373 @@
+            )
+            self.__set_visible(True)
+            self.__set_str_state('name', Path(self.image_source).stem)
+            self.__set_str_state('type', self.image_type)
+        
+        elif len(image_config):
+            # Setting default image_config in json image layer
+            if 'channel' not in image_config:
+                # Setting channel to 0 for image
+                self.__set_image_channel(0)
+                
+            if 'shaderControls' not in image_config:
+                self.__set_shader_control(
+                    {
+                        "normalized": {
+                            "range": [0, 600]
+                        }
+                    }
+                )
+                
+            if 'visible' not in image_config:
+                self.__set_visible(True)
+                
+            if 'name' not in image_config:
+                try:
+                    channel = self.layer_state['localDimensions']["c'"][0]
+                
+                except KeyError:
+                    channel = ''
+                self.__set_str_state('name', Path(self.image_source).stem + f"_{channel}")
+                
+            if 'type' not in image_config:
+                self.__set_str_state('type', self.image_type)
+    
+    def update_state(self, image_config:dict) -> None:
+        """
+        Set default values for the image.
+        
+        Parameters
+        ------------------------
+        image_config: dict
+            Dictionary with the image configuration. Similar to self.image_config
+            e.g.: image_config = {
+                'type': 'image', # Optional
+                'source': 'image_path',
+                'channel': 0, # Optional
+                'name': 'image_name', # Optional
+                'shader': {
+                    'color': 'green',
+                    'emitter': 'RGB',
+                    'vec': 'vec3'
+                },
+                'shaderControls': { # Optional
+                    "normalized": {
+                        "range": [0, 600]
+                    }
+                }
+            }
+        """
+        
+        for param, value in image_config.items():
+            if param in ['type', 'source', 'name']:
+                self.__set_str_state(param, value)
+                
+            if param in ['visible']:
+                self.__set_boolean_state(param, value)
+            
+            if param == 'shader':
+                self.__set_shader(
+                    self.__create_shader(
+                        value
+                    )
+                )
+                
+            if param == 'channel':
+                self.__set_image_channel(value)
+                
+            if param == 'shaderControls':
+                self.__set_shader_control(value)
+        
+        self.set_default_values(image_config)
+    
+    def __set_str_state(self, param:str, value:str) -> None:

```

Comment:
I don't think you need to do this much type validation.  Python will raise a sensible expression if casting to `str` doesn't work, I think.

Reply:
s/expression/exception/